### PR TITLE
Make Pallet ModuleId and LockIdentifier Configurable

### LINKED ISSUE #5149: `MODULE_ID` should be configurable
Some pallets introduce a `MODULE_ID` which is hard-coded into the runtime.

These are usually used as a seed for generating a unique account id for those pallets to hold funds (like a POT).

This pattern is not flexible, and immediately makes it so that these pallets could not be made instantiable.

It would be better to have these as configuration traits, and even better, to have them auto-populated based on the `ModuleToString`: https://github.com/paritytech/substrate/issues/5148

### PR DESCRIPTION
**Pull request by emostov**
Closes #5149 

This PR aims to switch `MODULE_ID` from being hardcoded into pallets over to being specified in runtime configuration.

polkadot companion: https://github.com/paritytech/polkadot/pull/1007

Below are the pallets changed and notes:

- ```treasury```
- ```elections```
    - no changes in runtime because ```elections``` does not appear to be incorporated currently.
- ```elections-phragrem```
    - please review the change to the test I made around line 1137. I changed it from ```assert_eq!(lock.id, MODULE_ID);``` to ```assert_eq!(lock.id, *b"phrelect");``` because ```MODULE_ID``` is no longer defined. However, I am not sure if the right value should be dynamic.
- ```evm```
    - no changes in runtime because ```evm``` does not appear to be incorporated currently.
- ```Society```

-----------------------------------------------------------
Before you submitting, please check that:

- [x] You added a brief description of the PR, e.g.:
  - What does it do?
  - What important points reviewers should know?
  - Is there something left for follow-up PRs?
- [x] You labeled the PR with appropriate labels if you have permissions to do so.
- [x] You mentioned a related issue if this PR related to it, e.g. `Fixes #228` or `Related #1337`.
- [x] You asked any particular reviewers to review. If you aren't sure, start with GH suggestions.
- [x] Your PR adheres [the style guide](https://github.com/paritytech/polkadot/wiki/Style-Guide)
  - In particular, mind the maximal line length.
  - There is no commented code checked in unless necessary.
  - Any panickers have a proof or removed.
- [ ] You bumped the runtime version if there are breaking changes in the **runtime**.
- [x] You updated any rustdocs which may have changed
- [x] Has the PR altered the external API or interfaces used by Polkadot? Do you have the corresponding Polkadot PR ready?

After you've read this notice feel free to remove it.
Thank you!

✄ -----------------------------------------------------------------------------

### CODE REVIEW

**Comment by shawntabrizi** [comment]
Looks about right, can you please check spacing issues with the lines you have introduced (we use tabs), and there seems to be compilation error, maybe just needs to merge master.

**Comment by emostov** [comment]
I merged in upstream/master and it looks like there still is a compilation error in some of the CI jobs. I did a release build and later ran all the non-ignored tests locally. Both seemed to finish without an issue. Not sure what else to try/look at regarding the compilation error.

Planning on having an accompanying Polkadot PR up within the next 24hrs as long as I can find a quick resolution to the CI errors.

File: frame/elections-phragmen/src/lib.rs [hunk 1]
```diff
@@ -320,7 +321,7 @@ decl_module! {
 
 			// lock
 			T::Currency::set_lock(
-				MODULE_ID,
+			T::ModuleId::get(),
```
Comment (shawntabrizi) [inline frame/elections-phragmen/src/lib.rs:324]: indent here
```suggestion
				T::ModuleId::get(),
```

File: frame/elections-phragmen/src/lib.rs [hunk 2]
```diff
@@ -1124,7 +1130,7 @@ mod tests {
 
 	fn has_lock(who: &u64) -> u64 {
 		let lock = Balances::locks(who)[0].clone();
-		assert_eq!(lock.id, MODULE_ID);
+		assert_eq!(lock.id, *b"phrelect");
```
Comment (bkchr) [inline frame/elections-phragmen/src/lib.rs:1133]:
```suggestion
		assert_eq!(lock.id, ElectionsPhragmenModuleId::get());
```

File: frame/elections/src/lib.rs [hunk 3]
```diff
@@ -147,6 +145,8 @@ type ApprovalFlag = u32;
 const APPROVAL_FLAG_LEN: usize = 32;
 
 pub trait Trait: frame_system::Trait {
+	type ModuleId: Get<LockIdentifier>;
```
Comment (bkchr) [inline frame/elections/src/lib.rs:148]: Docs.

Comment (emostov) [reply frame/elections/src/lib.rs]: Added a doc comment ```/// Identifier for the elections-phragmen pallet's lock``` and also an equivalent doc comment for ```elections```.

File: frame/elections/src/lib.rs [hunk 4]
```diff
@@ -379,6 +380,8 @@ decl_module! {
 		/// The chunk size of the approval vector.
 		const APPROVAL_SET_SIZE: u32 = APPROVAL_SET_SIZE as u32;
 
+		const MouduleId: LockIdentifier = T::ModuleId::get();
```
Comment (4meta5) [inline frame/elections/src/lib.rs:383]:
```suggestion
		const ModuleId: LockIdentifier = T::ModuleId::get();
```

File: frame/evm/src/lib.rs [hunk 5]
```diff
@@ -189,6 +189,8 @@ decl_module! {
 		type Error = Error<T>;
 
 		fn deposit_event() = default;
+		
+		const MouduleId: ModuleId = T::ModuleId::get();
```
Comment (4meta5) [inline frame/evm/src/lib.rs:193]:
```suggestion
		const ModuleId: ModuleId = T::ModuleId::get();
```

**Comment by bkchr** [comment]
Please merge master to fix the conflict.

**Comment by emostov** [comment]
Before this gets merged, should I take any steps to ensure that [the polkadot companion PR](https://github.com/paritytech/polkadot/pull/1007) can be merged in quick succession?

**Comment by bkchr** [comment]
> Before this gets merged, should I take any steps to ensure that [the polkadot companion PR](https://github.com/paritytech/polkadot/pull/1007) can be merged in quick succession?

Can we as push to your branch?

**Comment by emostov** [comment]
> > Before this gets merged, should I take any steps to ensure that [the polkadot companion PR](https://github.com/paritytech/polkadot/pull/1007) can be merged in quick succession?
> 
> Can we as push to your branch?

I think you are asking if you can push to my branch? If so, the answer is: yes.

**Pull request merged**